src/extra/widgets/animbtn/lv_animbtn.h

```diff
@@ -0,0 +1,118 @@
+/**
+ * @file lv_animbtn.h
+ *
+ */
+
+#ifndef LV_ANIMBTN_H
+#define LV_ANIMBTN_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "../../../lvgl.h"
+
+#if LV_USE_RLOTTIE != 0 && LV_USE_ANIMBTN != 0
+
+/*********************
+ *      DEFINES
+ *********************/
+typedef enum {
+    LV_ANIMBTN_STATE_RELEASED,
+    LV_ANIMBTN_STATE_PRESSED,
+    LV_ANIMBTN_STATE_DISABLED,
+    LV_ANIMBTN_STATE_CHECKED_RELEASED,
+    LV_ANIMBTN_STATE_CHECKED_PRESSED,
+    LV_ANIMBTN_STATE_CHECKED_DISABLED,
+    _LV_ANIMBTN_STATE_NUM,
+} lv_animbtn_state_t;
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+/*State status for anim button*/
+typedef struct {
+    size_t first_frame;
+    size_t last_frame;
+    size_t control; /* A lv_rlottie_ctrl_t instance and only checking Forward/Backward and Loop here */
```

Comment:
Really size_t is the appropriate type here?

Reply:
It's a classic chicken and egg here. I can't declare the type as `lv_rlottie_ctrl_t` since it's an enum and you can't forward declare an enum. If I include `lv_rlottie.h`, it includes `lvgl.h` which in turns include `lv_animbtn.h`. Since all files are protected by ifdef/define guard, so in the end, it does not work either. 

A better design would be to recreate the control we actually support making sure they match those of `lv_rlottie_ctrl_t`